cli/channels.go

```diff
@@ -5,200 +5,220 @@ package cli
+
+Examples:
+  channels all get <domain_id> <user_auth_token>                   				# Get all entities
+  channels <channel_id> get <domain_id> <user_auth_token>          				# Get specific entity
+  channels <channel_id> create <JSON_channel> <domain_id> <user_auth_token> 	# Create entity
+  channels <channel_id> update <JSON_string> <domain_id> <user_auth_token>  	# Update entity
+  channels <channel_id> delete <domain_id> <user_auth_token>      				# Delete entity
+  channels <channel_id> enable <domain_id> <user_auth_token>      				# Enable entity
+  channels <channel_id> disable <domain_id> <user_auth_token>     				# Disable entity
+  channels <channel_id> users <domain_id> <user_auth_token>       				# List entity users`,
 
-var cmdChannels = []cobra.Command{
-	{
-		Use:   "create <JSON_channel> <domain_id> <user_auth_token>",
-		Short: "Create channel",
-		Long:  `Creates new channel and generates it's UUID`,
 		Run: func(cmd *cobra.Command, args []string) {
-			if len(args) != 3 {
+			if len(args) < 2 {
 				logUsageCmd(*cmd, cmd.Use)
 				return
 			}
-
-			var channel smqsdk.Channel
-			if err := json.Unmarshal([]byte(args[0]), &channel); err != nil {
-				logErrorCmd(*cmd, err)
-				return
+			channelParams := args[0]
+			operation := args[1]
+			opArgs := args[2:]
+
+			switch operation {
+			case create:
+				handleCreate(cmd, channelParams, opArgs)
+			case get:
+				handleGet(cmd, channelParams, opArgs)
+			case update:
+				handleUpdate(cmd, channelParams, opArgs)
+			case delete:
+				handleDelete(cmd, channelParams, opArgs)
+			case enable:
+				handleEnable(cmd, channelParams, opArgs)
+			case disable:
+				handleDisable(cmd, channelParams, opArgs)
+			case users:
+				handleUsers(cmd, channelParams, opArgs)
+			default:
+				logErrorCmd(*cmd, fmt.Errorf("unknown operation: %s", operation))
 			}
+		},
+	}
 
-			channel, err := sdk.CreateChannel(cmd.Context(), channel, args[1], args[2])
-			if err != nil {
-				logErrorCmd(*cmd, err)
-				return
-			}
+	return cmd
+}
 
-			logJSONCmd(*cmd, channel)
-		},
-	},
-	{
-		Use:   "get [all | <channel_id>] <domain_id> <user_auth_token>",
-		Short: "Get channel",
-		Long: `Get all channels or get channel by id. Channels can be filtered by name or metadata.
-		all - lists all channels
-		<channel_id> - shows client with provided <channel_id>`,
+func handleCreate(cmd *cobra.Command, channelID string, args []string) {
+	if len(args) != 2 {
+		logUsageCmd(*cmd, usageCreate)
+		return
+	}
 
-		Run: func(cmd *cobra.Command, args []string) {
-			if len(args) != 3 {
-				logUsageCmd(*cmd, cmd.Use)
-				return
-			}
-			metadata, err := convertMetadata(Metadata)
-			if err != nil {
-				logErrorCmd(*cmd, err)
-				return
-			}
-			pageMetadata := smqsdk.PageMetadata{
-				Name:     "",
-				Offset:   Offset,
-				Limit:    Limit,
-				Metadata: metadata,
-			}
+	var channel smqsdk.Channel
+	if err := json.Unmarshal([]byte(channelID), &channel); err != nil {
```

Comment:
This is not corect, creating a channel should allow fro metadata and name in the json string